internal/feature/flags.go

```diff
@@ -0,0 +1,72 @@
+package feature
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Defaults map[string]bool
+
+// All feature flags need to be defined here with their default values
+var flags = Defaults{
+	"events": false,
+}
+
+type Feature interface {
+	IsEnabled(flag string) (bool, error)
+	Enable(flag string) error
+	Disable(flag string) error
+}
+
+func ProvideFeature() Feature {
+	return newStaticMapFeature(flags)
+}
+
+func newStaticMapFeature(defaults Defaults) *staticMapFeature {
+	// copy map so we don't rely on global state
+	newMap := map[string]bool{}
+	for key, value := range defaults {
+		newMap[key] = value
+	}
+	return &staticMapFeature{flags: newMap, mu: &sync.Mutex{}}
+}
+
+type staticMapFeature struct {
+	flags map[string]bool
+	mu    *sync.Mutex
```

Comment:
nit: this doesn't need to be a pointer since the method receivers are pointers (which then also means this field doesn't need to be initialized)